pkg/steps/pre_publish_images.go

```diff
@@ -0,0 +1,155 @@
+package steps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+
+	imageapi "github.com/openshift/api/image/v1"
+	"github.com/openshift/ci-operator/pkg/api"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	coreapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PrePublishOutputImageTagStep will ensure that a tag exists
+// in the named ImageStream that resolves to the built
+// pipeline image
+type prePublishOutputImageTagStep struct {
+	config    api.PrePublishOutputImageTagStepConfiguration
+	istClient imageclientset.ImageStreamTagsGetter
+	isClient  imageclientset.ImageStreamsGetter
+	jobSpec   *api.JobSpec
+	tag       string
+}
+
+type prepublishConfig struct {
+	namespace, name, tag string
+}
+
+func (s *prePublishOutputImageTagStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return nil, nil
+}
+
+func (s *prePublishOutputImageTagStep) Run(ctx context.Context, dry bool) error {
```

Comment:
Is there an opportunity to DRY this out with the other publish step?

Reply:
Definitely. I was going to revisit that. I extract what I can out into some helper functions.